src/kemal/common_error_handler.cr

```diff
@@ -0,0 +1,17 @@
+
+class Kemal::CommonErrorHandler < HTTP::Handler
+  INSTANCE = new
+
+  def call(context)
+    begin
+      call_next context
+    rescue ex : Kemal::Exceptions::RouteNotFoundException
+      Kemal.config.logger.write("Exception: #{ex.to_s}: #{ex.message}\n")
+      return render_404(context)
+    rescue ex
+      Kemal.config.logger.write("Exception: #{ex.to_s}: #{ex.message}\n")
+      return render_500(context, ex.to_s)
+    end
+  end
+
+end
```

Comment:
Well that's not very common but we're rescuing from the exception with an intent to notify the user.